packages/typescript/internal/ts_project.bzl

```diff
@@ -641,26 +640,19 @@ def ts_project_macro(
                 # Users get this dependency transitively from @bazel/typescript
                 # but that's our own code, so we don't.
                 "@npm//protobufjs",
+                "@npm//typescript",
```

Comment:
Is there any way to not remove an ability to provide a custom typescript compiler package? In our company, we are required to use [ttypescript](https://github.com/cevek/ttypescript) in some places.

If it will be too much for this PR, will you accept an external contribution for such a feature, or we just should patch it on our end?

Reply:
There would definitely be a way to still support cusotm typescript providers. We would just need to replace `import * as ts from 'typescript` with a `const ts = require(SOME_ARGUMENT)`. I'd love to leave that as a followup if possible

Reply:
So I have the logic here right to support ttypescript, but it turns out that they have a bug in which they aren't patching all the typescript APIs that are available. I've filed a bug for them: https://github.com/cevek/ttypescript/issues/110